### ISSUE TITLE
🐜 [Bug][1.7.1]Player will be revived on Gurabashi arena after they died in dungeon Uldaman

### ISSUE THREAD

**Issue by den13501** [vmangos/core#1492]
## 🐜 Bug report - Progression

When player dead in Uldaman, they should be revived on Badlands, not Gurabashi arena, this issue only happen on old patch of server like 1.7.1.
### Steps to reproduce

1. `.tele uldaman`
2. `.die` to kill yourself and press release spirit

### Version & Environment

Client Version: 1.7.1.4695

Commit Hash:  

OS Client: Windows 10
OS Server: Windows server 2008 R2

### Crashlog

- None

**Comment by den13501** [comment]
I think why I would be teleport to Gurabashi  but not Badlands, check WorldSafeLocs.dbc , the alliance id of graveyard in badland is 8 in client 1.12, but id 8 is Gurabashi arena in client 1.11 and all old client.
![image](https://user-images.githubusercontent.com/58270741/167465379-3152c1e8-df65-45bc-8862-21bc1fad5476.png)

**Comment by den13501** [comment]
# Hack fix
1. Added a new definated  row in worldsafeloc.dbc data.(for example, mine is ID=9)
2. Insert these rows into mangos db
```
INSERT INTO `game_graveyard_zone` (`id`, `ghost_zone`, `faction`, `build_min`) VALUES (9, 3, 0, 4544);
INSERT INTO `game_graveyard_zone` (`id`, `ghost_zone`, `faction`, `build_min`) VALUES (9, 1337, 0, 4544);
INSERT INTO `world_safe_locs_facing` (`id`, `orientation`) VALUES (9, 2.79253);
```
3. Replace original dbc with modified WorldSafeLocs.dbc
[WorldSafeLocs.zip](https://github.com/vmangos/core/files/8654642/WorldSafeLocs.zip)

**Comment by Daribon** [comment]
Can confirm Stranglethorn was replaced by Badlands graveyard in worldsafelocs.
Worldsafeloc 8 STV arena was never any actual graveyard, remnant from pre graveyard system.

Worldsafelocs entry 8 was changed to the text reuse by patch 1.9.0:
![WDBX_Editor_OFArR7qG24](https://user-images.githubusercontent.com/6137576/167487393-4ef16c71-d1ce-472d-a2b5-6ece4a21a5e0.png)

Worldsafelocs entry 8 gets reused by patch 1.12.0 to be the new badlands graveyard added in 1.12.0:
![WDBX_Editor_4hBTZiqBYZ](https://user-images.githubusercontent.com/6137576/167487650-a76bdf9a-89b1-4f51-a283-38377e076216.png)

I'm not surprised we didn't find this earlier, since there is absolutly no mention of it in patch notes that northern badlands graveyard was added in patch 1.12.0, but both terrain and worldsafelocs was modified in 1.12.0 to add it.

Patch 1.11.2:
![WoW_XMnPI5ZdHA](https://user-images.githubusercontent.com/6137576/167489301-600de193-03c2-4726-ac14-ce8f5f217a78.png)

Patch 1.12.1:
![WoW_clyBTtKM50](https://user-images.githubusercontent.com/6137576/167489344-f846a60f-f6c6-4134-895f-60b9f55ec560.png)

TL;DR, the actual issue is that Vmangos treats worldsafeloc 8 as a graveyard on all patches, when it only becomes a graveyard in patch 1.12.0.

**Comment by Daribon** [comment]
I've made a fix for this, will commit PR soon with many fixes related to graveyards.

**Comment by den13501** [comment]
Thanks @Daribon , but I found this issue is still existed after apply this commit.
Test on server set patch as 1.8.

**Comment by Daribon** [comment]
> Thanks @Daribon , but I found this issue is still existed after apply this commit. Test on server set patch as 1.8.

I have not fixed it yet, the commit I made recently was just to correct positions.
There will be another commit sometime soon which fixes plenty of graveyards around Azeroth.
